src/Uno.UI/UI/Xaml/Controls/SplitView/SplitView.core.cs

```diff
@@ -0,0 +1,618 @@
+			// passing in false for 'useTransitions.'
+			UpdateVisualState(false);
+		}
+	}
+
+	// Uno Doc: CSplitView's OnApplyTemplate
+	private void COnApplyTemplate()
+	{
+		UnregisterEventHandlers();
+
+		// Clear any hold-overs from the previous template.
+		m_spPaneClipRectangle = null;
+		m_spContentRoot = null;
+		m_spPaneRoot = null;
+		m_spLightDismissLayer = null;
+
+		base.OnApplyTemplate();
+
+		DependencyObject spTemplateChild = GetTemplateChild("PaneClipRectangle");
+		m_spPaneClipRectangle = spTemplateChild as RectangleGeometry;
+
+		spTemplateChild = GetTemplateChild("ContentRoot");
+		m_spContentRoot = spTemplateChild as UIElement;
+
+		spTemplateChild = GetTemplateChild("PaneRoot");
+		m_spPaneRoot = spTemplateChild as UIElement;
+
+		spTemplateChild = GetTemplateChild("LightDismissLayer");
+		m_spLightDismissLayer = spTemplateChild as UIElement;
+
+		RegisterEventHandlers();
+		UpdateTemplateSettings();
+		UpdateVisualState(true);
+	}
+
+	protected override Size MeasureOverride(Size availableSize)
+	{
+		// Measure the pane content so that we can use the desired size in cases
+		// where open pane length is set to Auto.
+		var value = Pane;
+		if (value is { })
+		{
+			var pPaneElement = value;
+
+			pPaneElement.Measure(availableSize);
+
+			m_paneMeasuredLength = pPaneElement.DesiredSize.Width;
+		}
+
+		var @out =  base.MeasureOverride(availableSize);
+
+		UpdateTemplateSettings();
+
+		return @out;
+	}
+
+	protected override Size ArrangeOverride(Size finalSize)
+	{
+		var newFinalSize = base.ArrangeOverride(finalSize);
+
+		if (m_spPaneClipRectangle is { })
+		{
+			Rect rect = new Rect( 0, 0, OpenPaneLength, newFinalSize.Height);
+			m_spPaneClipRectangle.Rect = rect;
+		}
+
+		return newFinalSize;
+	}
+
+	private bool IsLightDismissible()
+	{
+		return DisplayMode is not (SplitViewDisplayMode.Inline or SplitViewDisplayMode.CompactInline);
+	}
+
+	private bool CanLightDismiss()
+	{
+		return IsPaneOpen && !m_isPaneClosingByLightDismiss && IsLightDismissible();
+	}
+
+	private double GetOpenPaneLength()
+	{
+		double openPaneLength = OpenPaneLength;
+
+		// Support Auto/NaN for open pane length to size to the pane content.
+		if (openPaneLength.IsNaN())
+		{
+			openPaneLength = m_paneMeasuredLength;
+		}
+
+		return openPaneLength;
+	}
+
+	private void TryCloseLightDismissiblePane()
+	{
+		global::System.Diagnostics.Debug.Assert(CanLightDismiss());
+
+		var core = this.GetContext();
+		SplitViewPaneClosingEventArgs spArgs = new();
+
+		// Uno Specific
+		// // Raise the closing event to give the app a chance to cancel.
+		// CEventManager *pEventManager = core->GetEventManager();
+		// if (pEventManager != nullptr)
+		// {
+		// 	pEventManager->Raise(
+		// 		EventHandle(KnownEventIndex::SplitView_PaneClosing),
+		// 		TRUE /*bRefire*/,
+		// 		this,
+		// 		spArgs,
+		// 		FALSE /*fRaiseSync*/
+		// 	);
+		// }
+		PaneClosing?.Invoke(this, spArgs);
```

Comment:
I think this should be dispatched?

Reply:
Chasing the function calls to the source, it doesn't seem like it's dispatched (I had the same thought too).

Reply:
@ramezgerges It should be. This is what `fRaiseSync` for.